libnavigation-core/src/main/java/com/mapbox/navigation/core/internal/trip/session/MapboxTripSession.kt

```diff
@@ -64,7 +65,8 @@ class MapboxTripSession(
     private val navigatorPredictionMillis: Long,
     private val navigator: MapboxNativeNavigator = MapboxNativeNavigatorImpl,
     threadController: ThreadController = ThreadController,
-    private val logger: Logger
+    private val logger: Logger,
+    private val accessToken: String?
```

Comment:
will it be possible to be **NULL**? I am asking this because I am thinking what the Line 477 will be when it's NULL and what will backend service return to us if we pass a null `access_token`?

Reply:
The access token sent here is from `NavigationOptions` which has it nullable. Having null accessToken in this case is ok because if we send null in line 477, the user won't be able to draw the guidance view. In order to make it non-null I will have to change it in `NavigationOptions` and I don't want to do that, as I am not sure what kind of regressions this might introduce.

Reply:
It can be tracked as a follow up issue. I did a quick check about the usage of `access_token` in `NavigationOptions`. 
I think we do have the `NULL` check when we initiate the `router` module in `MapboxNavigation`
```kotlin
MapboxModuleType.NavigationOffboardRouter -> arrayOf(
                String::class.java to (accessToken
                    ?: throw RuntimeException(MAPBOX_NAVIGATION_TOKEN_EXCEPTION_OFFBOARD_ROUTER)),
                Context::class.java to navigationOptions.applicationContext,
                UrlSkuTokenProvider::class.java to MapboxNavigationAccounts.getInstance(navigationOptions.applicationContext)
            )
```

To me it seems the `access_token` in `NavigationOptions` should be a non-null property, isn't it?
cc: @Guardiola31337 , @kmadsen , @LukasPaczos

Reply:
I totally agree that it should be NON-NULL, but I think I am missing the discussion as to why was it made `NULLABLE` in the first place.

Reply:
The only places in the code that use the token are the routers. Since they are modular, the developer could replace the implementations and not actually need the token, that's why it's nullable.